src/main/java/games/strategy/triplea/delegate/MoveDelegate.java

```diff
@@ -363,20 +363,19 @@ static Change giveBonusMovementToUnits(final PlayerID player, final GameData dat
         }
         int bonusMovement = Integer.MIN_VALUE;
         final Collection<Unit> givesBonusUnits = new ArrayList<>();
-        final Predicate<Unit> givesBonusUnit = Match.allOf(Matches.alliedUnit(player, data),
-            Matches.unitCanGiveBonusMovementToThisUnit(u));
+        final Predicate<Unit> givesBonusUnit = Matches.alliedUnit(player, data)
+            .and(Matches.unitCanGiveBonusMovementToThisUnit(u));
         givesBonusUnits.addAll(Matches.getMatches(t.getUnits().getUnits(), givesBonusUnit));
         if (Matches.unitIsSea().test(u)) {
-          final Predicate<Unit> givesBonusUnitLand = Match.allOf(givesBonusUnit, Matches.unitIsLand());
+          final Predicate<Unit> givesBonusUnitLand = givesBonusUnit.and(Matches.unitIsLand());
           final List<Territory> neighbors =
               new ArrayList<>(data.getMap().getNeighbors(t, Matches.territoryIsLand()));
           for (final Territory current : neighbors) {
             givesBonusUnits.addAll(Matches.getMatches(current.getUnits().getUnits(), givesBonusUnitLand));
           }
         } else if (Matches.unitIsLand().test(u)) {
-          final Predicate<Unit> givesBonusUnitSea = Match.allOf(givesBonusUnit, Matches.unitIsSea());
-          final List<Territory> neighbors =
-              new ArrayList<>(data.getMap().getNeighbors(t, Matches.territoryIsWater()));
+          final Predicate<Unit> givesBonusUnitSea = givesBonusUnit.and(Matches.unitIsSea());
+          final Set<Territory> neighbors = data.getMap().getNeighbors(t, Matches.territoryIsWater());
```

Comment:
Don't know if this is an issue, but the defensive copy may have been there to prevent a `ConcurrentModificationException`.  In some cases, `GameMap#getNeighbors()` may not return a defensive copy.  In that case, if the returned `Set` is modified during the iteration below, a `ConcurrentModificationException` will be thrown.

Reply:
Good Point, will fix that

Reply:
Fix it, changed the wrapping type to a hash set though